### PR TITLE
Promiscuous mode checkin

### PR DESCRIPTION
**Pull request by VivekUppunda**
This set of changes adds promiscuous mode support to nrf Wi-Fi driver.

### CODE REVIEW

File: drivers/wifi/nrf700x/src/wifi_mgmt.c [hunk 1]
```diff
@@ -791,11 +797,19 @@ int nrf_wifi_mode(const struct device *dev,
 
 	} else {
 		mode->mode = def_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx]->mode;
-
+		/**
+		 * This is a work-around to handle current UMAC mode handling.
+		 * This might be removed in future versions when UMAC has more space.
+		 */
 #if defined(CONFIG_NRF700X_RAW_DATA_TX) || defined(CONFIG_NRF700X_RAW_DATA_RX)
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/src/wifi_mgmt.c:804]:
```suggestion
#ifdef(CONFIG_NRF700X_RAW_DATA_TX)
```

File: drivers/wifi/nrf700x/src/wifi_mgmt.c [hunk 2]
```diff
@@ -791,11 +797,19 @@ int nrf_wifi_mode(const struct device *dev,
...
 		if (def_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx]->txinjection_mode == true) {
 			mode->mode ^= NRF_WIFI_TX_INJECTION_MODE;
 		}
+#endif
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/src/wifi_mgmt.c:808]:
```suggestion
#endif /* CONFIG_NRF700X_RAW_DATA_TX */
```

File: drivers/wifi/nrf700x/src/wifi_mgmt.c [hunk 3]
```diff
@@ -791,11 +797,19 @@ int nrf_wifi_mode(const struct device *dev,
...
+#ifdef CONFIG_NRF700X_PROMISC_DATA_RX
+		if (def_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx]->promisc_mode == true) {
+			mode->mode ^= NRF_WIFI_PROMISCUOUS_MODE;
+		}
 #endif
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/src/wifi_mgmt.c:813]:
```suggestion
#endif /* CONFIG_NRF700X_PROMISC_DATA_RX */
```

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

**Review by sachinthegreen — APPROVED** [review]

File: drivers/wifi/nrf700x/Kconfig [hunk 4]
```diff
@@ -84,6 +84,10 @@ config NRF700X_RAW_DATA_RX
 	bool "Enable RAW RX sniffer operation in the driver"
 	depends on NRF700X_SYSTEM_MODE
 
+config NRF700X_PROMISC_DATA_RX
+	bool "Enable RAW promiscuous RX sniffer operation in the driver"
```
Comment (krish2718) [inline drivers/wifi/nrf700x/Kconfig:88]:
```suggestion
	bool "Enable promiscuous RX sniffer operation in the driver"
```

File: drivers/wifi/nrf700x/src/wifi_mgmt.c [hunk 5]
```diff
@@ -791,19 +797,27 @@ int nrf_wifi_mode(const struct device *dev,
...
+		 */
+#ifdef CONFIG_NRF700X_RAW_DATA_TX
 		if (def_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx]->txinjection_mode == true) {
 			mode->mode ^= NRF_WIFI_TX_INJECTION_MODE;
 		}
-#endif
+#endif /* CONFIG_NRF700X_RAW_DATA_TX */
+#ifdef CONFIG_NRF700X_PROMISC_DATA_RX
+		if (def_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx]->promisc_mode == true) {
+			mode->mode ^= NRF_WIFI_PROMISCUOUS_MODE;
```
Comment (krish2718) [inline drivers/wifi/nrf700x/src/wifi_mgmt.c:811]: This is the only place where the define is used as standalone, so, do we really need a separate define for promiscuous ? Can't we use RAW_DATA_RX for both and rely on `mode` runtime setting?

Comment (VivekUppunda) [reply drivers/wifi/nrf700x/src/wifi_mgmt.c:811]: It has other checks specifically for Promiscuous mode. One of the ask initially was to keep different implementations completely code configurable during build. It was as per that discussion that Monitor, Promiscuous and TX Injection modes are kept separate.

Comment (krish2718) [reply drivers/wifi/nrf700x/src/wifi_mgmt.c:811]: > It has other checks specifically for Promiscuous mode

Ok, in the driver?

> It was as per that discussion that Monitor, Promiscuous and TX Injection modes are kept separate.

Anyways, I see a lot of duplication b/w them, so, we should re-discuss, but we can proceed with the PR.

**Review by kapbh — APPROVED** [review]

**Pull request merged**